### PR TITLE
refactor: entry card

### PR DESCRIPTION
**Pull request by didd**
## Description

- Refactor entry card

## Checklist

- [X] I have read the **README** document
- [X] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [X] All new and existing tests passed
- [X] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

**Review by quininez — DISMISSED** [review]

File: ui/design-system-components/src/components/Entry/EntryCard/index.tsx [hunk 1]
```diff
@@ -1,130 +1,216 @@
...
+                content={rest.slateContent}
+                handleMentionClick={rest.onMentionClick}
+                handleTagClick={rest.onTagClick}
+                handleLinkClick={url => {
+                  rest.navigateTo?.({ getNavigationUrl: () => url });
+                }}
+              />
+            ) : (
+              rest.sortedContents.map(item => (
+                <Extension
```
Comment (SeverS) [inline ui/design-system-components/src/components/Entry/EntryCard/index.tsx]: We should not have this <Extension /> here, we should have {props.children} instead and control the extension from outside. I'm working to replace this extension component with another one which will be in it's own library (like the feed lib)

**Review by SeverS — CHANGES_REQUESTED** [review]
The <Extension /> should be replaced with {children} and adjust the new changes

File: ui/design-system-components/src/components/Entry/EntryCard/card-actions.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        </Button>
+      </Stack>
+    );
+  }
+
+  return (
+    <Stack direction="row" align="center" justify="end" spacing="gap-x-4" padding="p-4">
+      <>{actionsRightExt}</>
+      <Anchor
+        href={`${repliesAnchorLink}/${profileId}`}
```
Comment (SeverS) [inline ui/design-system-components/src/components/Entry/EntryCard/card-actions.tsx]: is this href right? shouldn't be ${beamId} or something?

File: ui/design-system-components/src/components/Entry/EntryCard/index.tsx [hunk 3]
```diff
@@ -1,130 +1,216 @@
...
+        </Button>
+      )}
+      {!hideActionButtons && (
+        <CardActions
+          profileId={entryData.author.id}
+          repliesAnchorLink={repliesAnchorLink}
+          disableActions={disableActions}
+          isModerated={isModerated}
+          actionsRightExt={actionsRightExt}
+          onReflect={onReflect}
```
Comment (SeverS) [inline ui/design-system-components/src/components/Entry/EntryCard/index.tsx:200]: isn't this prop (`onReflect`) used to navigate to a beam's page reflections section?

Comment (didd) [reply ui/design-system-components/src/components/Entry/EntryCard/index.tsx:200]: If you saw the comment at the top of the card-actions component I have a todo to refactor the component :)

File: ui/design-system-components/src/components/Entry/EntryCard/card-actions.tsx (hunk 2 above)
Comment (didd) [reply ui/design-system-components/src/components/Entry/EntryCard/card-actions.tsx]: Yes, it should be the `beamId`. Btw I have a todo for this component to refactor it that I didn't bother much about it.

File: ui/design-system-components/src/components/Entry/EntryCard/index.tsx (hunk 3 above)
Comment (SeverS) [reply ui/design-system-components/src/components/Entry/EntryCard/index.tsx:200]: oh, oky, sorry I missed that

**Review by SeverS — APPROVED** [review]

**Pull request merged**